kratos/sources/communicator.cpp

```diff
@@ -133,6 +133,22 @@ void Communicator::SetNumberOfColors(SizeType NewNumberOfColors)
     }
 }
 
+void Communicator::AddColors(SizeType NumberOfAddedColors)
+{
+    if (NumberOfAddedColors < 1)
+        return;
+
+    mNumberOfColors += NumberOfAddedColors;
+    MeshType mesh;
+
+    for (IndexType i = 0; i < NumberOfAddedColors; i++)
+    {
+        mLocalMeshes.push_back(Kratos::make_shared<MeshType>(mesh.Clone()));
+        mGhostMeshes.push_back(Kratos::make_shared<MeshType>(mesh.Clone()));
+        mInterfaceMeshes.push_back(Kratos::make_shared<MeshType>(mesh.Clone()));
+    }
+}
+
```

Comment:
Just mentioning that I think this belongs to #7192. Can you please revert?

Reply:
sorry don't know why it follows along, but i reverted it.